scripts/ci/determine-tests.mjs

```diff
@@ -176,14 +176,17 @@ const changedFiles = getChangedFiles(base, head).split(EOL);
 const outputNameTestsToRun = process.argv[2];
 const outputNameShardIndex = process.argv[3];
 const outputNameShardTotal = process.argv[4];
+const runAllTests = process.argv[5] === 'true';
 const projectRoot = process.env.projectRoot;
 const atomicSourceComponents = join('packages', 'atomic', 'src', 'components');
 
 try {
   const testFiles = findAllTestFiles(atomicSourceComponents);
   const testDependencies = createTestFileMappings(testFiles, projectRoot);
-  const testsToRun = determineTestFilesToRun(changedFiles, testDependencies);
-  if (testsToRun === '') {
+  const testsToRun = runAllTests
+    ? testFiles.join(' ')
+    : determineTestFilesToRun(changedFiles, testDependencies);
+  if (testsToRun === '' && !runAllTests) {
     throw new NoRelevantChangesError();
   }
```

Comment:
This is hard to parse imo, I would put it this way very clearly and exit with an error if runAllTests is true before running all other functions. So at the very beginning of the try I would if check runAllTests and immediately throw. You can then catch it with the other 'run all tests' errors.

```
class RunAllTestsInMergeQueueError extends Error {
  constructor() {
    super(
      `In a merge queue. Running all tests.`
    );
    this.name = 'RunAllTestsInMergeQueueError';
  }
}
try {
if(runAllTests){
throw new RunAllTestsInMergeQueue();
}

  const testFiles = findAllTestFiles(atomicSourceComponents);
  const testDependencies = createTestFileMappings(testFiles, projectRoot);
  const testsToRun = determineTestFilesToRun(changedFiles, testDependencies);
  if (testsToRun === '') {
    throw new NoRelevantChangesError();
  }
  ...
  
  catch (error) {
...

  if (
    error instanceof DependentPackageChangeError ||
    error instanceof PackageJsonChangeError ||
    error instanceof RunAllTestsInMergeQueueError
  ) {
```

Reply:
Good point, I'll improve that